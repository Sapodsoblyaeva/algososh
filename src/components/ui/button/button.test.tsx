import { render, screen } from "@testing-library/react";
import { Button } from "./button";
import userEvent from "@testing-library/user-event";

describe("button test", () => {
  it("button  disabled", () => {
    render(<Button disabled={true} />);
    
    const button = screen.getByTestId("button");

    expect(button).toHaveAttribute("disabled");
    expect(button).toMatchSnapshot();
  });
  it("button work of loader", () => {
    render(<Button isLoader={true} />);

    const button = screen.getByTestId("button");

    expect(button).toHaveClass("loader");
    expect(button).toMatchSnapshot();
  });
  it("button with text", () => {
    render(<Button text="click" />);

    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("click");

    expect(button).toMatchSnapshot();
  });
  it("button without text", () => {
    render(<Button />);
    
    const button = screen.getByTestId("button");

    expect(button).not.toHaveTextContent("click");
    expect(button).toMatchSnapshot();
  });
  it("button click", () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick} />);

    const button = screen.getByTestId("button");
    userEvent.click(button);
    
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
