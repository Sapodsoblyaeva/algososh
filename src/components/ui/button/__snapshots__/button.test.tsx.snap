// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test button  disabled 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`button test button with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  >
    click
  </p>
</button>
`;

exports[`button test button without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`button test button work of loader 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;
